libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/snap/SnapToRoute.java

```diff
@@ -163,8 +163,18 @@ private static Point getUpcomingLegPoint(RouteProgress routeProgress) {
       return null;
     }
 
-    RouteLeg upcomingLeg = routeProgress.directionsRoute().legs().get(routeProgress.legIndex() + 1);
-    if (upcomingLeg.steps() == null || upcomingLeg.steps().size() < 1) {
+    // Search upcoming leg with at least two steps.
+    RouteLeg upcomingLeg = null;
+    List<RouteLeg> upcomingLegs = routeProgress.directionsRoute().legs()
+            .subList(routeProgress.legIndex() + 1, routeProgress.directionsRoute().legs().size());
+    for (RouteLeg leg : upcomingLegs) {
+      if (leg.steps() != null && leg.steps().size() >= 2) {
```

Comment:
Wouldn't it be easier to use a for(i= routeProgress.legIndex() + 1 ; ...)? That way we don't need sublists.

In general since returning null is acceptable, I am not sure if we shouldn't just return null if the next leg has no steps or < 2 steps? But I guess both approaches would be fine as this seems to be an edge case either way.

Reply:
> Wouldn't it be easier to use a for(i= routeProgress.legIndex() + 1 ; ...)? That way we don't need sublists.

Sure. Make sense!

> In general since returning null is acceptable, I am not sure if we shouldn't just return null if the next leg has no steps

If we return `NULL` the bearing is set to `0`, this means the puck and camera is turn around to zero position, and will turn back for next step.

this also happens on the last position. Maybe we should store the latest bearing and return re-use them instead of return a `0` bearing. In this case we don't need the loop-stuff.

What do you think?

Reply:
I like the idea of fallback to the real location. But this is often not route matched and we have also the issue #63. So I would implement the last bearing usage for now, but I think we can improve this later after some other topics to the location.